compiler-plugins/kapt/src/org/jetbrains/kotlin/kapt/idea/KaptProjectResolverExtension.kt

```diff
@@ -31,7 +32,15 @@ class KaptProjectResolverExtension : AbstractProjectResolverExtension() {
         private val LOG = Logger.getInstance(KaptProjectResolverExtension::class.java)
     }
 
-    override fun getExtraProjectModelClasses() = setOf(KaptGradleModel::class.java)
+    override fun getExtraProjectModelClasses(): Set<Class<KaptGradleModel>> {
+        val isAndroidPluginRequestingKotlinGradleModelKey = Key.findKeyByName("IS_ANDROID_PLUGIN_REQUESTING_KAPT_GRADLE_MODEL_KEY")
+        if (isAndroidPluginRequestingKotlinGradleModelKey != null && resolverCtx.getUserData(isAndroidPluginRequestingKotlinGradleModelKey) != null) {
+            return emptySet()
+        }
+
+        return setOf(KaptGradleModel::class.java)
```

Comment:
The long term plan is to separate responsibilities of Android and Kotlin plugins at both Gradle plugin level and the IDE plugin level. Right now both plugins, unfortunately, know about each other. When done Android modules should be set up by the Android plugin and non-Android modules by the platform + koltin plugin. 

At this moment, when a new version of Android plugin is present it takes over requesting `KoltinGradleModel` and, with this change, `KaptGradleModel` for both Android and non-Android modules. However, for Android modules it requests a parameterised model. This is obviously not a good design choice, but we were late with more elaborate solution for 1.4.30 (in 14.30 this solution works for KoltinGradleModel only). 

I'll have another PR which safeguards IDEA from possible Android plugin misbehavior by moving the responsibility to request models for non-Android modules back to the kotlin plugin . However, since we already have this behavior in kotlin 1.4.30 for koltin models it needs to be backward compatible. 

You can see it works now for Kotlin models right now here: 

https://android.googlesource.com/platform/tools/adt/idea/+/refs/heads/mirror-goog-studio-master-dev/android/src/com/android/tools/idea/gradle/project/sync/idea/svs/AndroidExtraModelProvider.kt